packages/cli/tests/unit/encodedFile.spec.ts

```diff
@@ -0,0 +1,67 @@
+import fs from 'fs';
+import { join } from "path";
+import tmp from 'tmp';
+
+import EncodedFile from '../../src/encodedFile';
+
+
+tmp.setGracefulCleanup();
+describe('EncodedFile', () => {
+  let tmpDir;
+  let fixtureDir;
+  beforeEach(() => {
+    tmpDir = tmp.dirSync({} as any).name
+    fixtureDir = join(__dirname, 'fixtures', 'encodedFile');
+  });
+
+  describe('on Windows', () => {
+    // The UTF16LE fixtures have Windows line endings.
+    beforeEach(() => {
+      fixtureDir = join(__dirname, 'fixtures', 'encodedFile', 'windows');
+    });
+
+    it('decodes a UTF-16 file', () => {
+      const ef = new EncodedFile(join(fixtureDir, 'utf16le.txt'));
+      const str = ef.toString();
+      expect(str[0]).toEqual('#');
+    });
+
+    it('writes an updated UTF-16 file', () => {
+      const prefix = 'utf16le';
+
+      const inputPath = join(fixtureDir, `${prefix}.txt`);
+      const actualPath = join(tmpDir, `${prefix}.actual.txt`);
+      const ef = new EncodedFile(inputPath);
+      const newString = '# New text\r\n' + ef.toString();
+      ef.write(newString, actualPath);
+
+      const expectedPath = join(fixtureDir, `${prefix}.expected.txt`);
+      const expected = fs.readFileSync(expectedPath);
+      const actual = fs.readFileSync(actualPath);
+      expect(actual).toEqual(expected);
+    });
+  });
+
+  it('reads a 7-bit ASCII file', () => {
+    const ef = new EncodedFile(join(fixtureDir, 'ascii.txt'));
+    const str = ef.toString();
+    expect(str[0]).toEqual('#');
+  });
+  
+  it('writes a 7-bit ASCII file', () => {
+    const expectedPath = join(fixtureDir, 'ascii.txt');
+    const ef = new EncodedFile(expectedPath);
+    const actualPath = join(tmpDir, 'ascii.actual.txt');
+    ef.write(ef.toString(), actualPath);
+    const expected = fs.readFileSync(expectedPath);
+    const actual = fs.readFileSync(actualPath);
+    expect(actual).toEqual(expected);
+  });
```

Comment:
You need to add some more cases here:
- UTF-8 with BOM,
- UTF-8 with Windows line endings,
- UTF-16 with Unix line endings.